### PR TITLE
Multiple bug fixes

### PR DESCRIPTION
**Pull request by LordGolias**
## What type of PR is this.
1. [x] Bug
2. [ ] Change
3. [ ] Enhancement

### What have you changed and why?
Information: multiple issues regarding the incorrect precedence of operations, resulting in different behavior than what is intended.

These issues were identified by [sqf-analyzer](https://marketplace.visualstudio.com/items?itemName=SQF-analyzer.sqf-analyzer).

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

Not yet, I can but would like to ensure this PR is acceptable before running manual tests.

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

This may have changed behavior in ways that were not intended, since the code was tested with the previously (likely incorrect) behavior.

### CODE REVIEW

File: A3A/addons/core/functions/init/fn_resourcecheck.sqf [hunk 1]
```diff
@@ -192,7 +192,7 @@ while {true} do
 			["TaskSucceeded", ["", format ["%1 Rebuilt",_nameX]]] remoteExec ["BIS_fnc_showNotification",[teamPlayer,civilian]];
 			sleep 2;
 			};
-		} forEach (destroyedSites - citiesX) select {sidesX getVariable [_x,sideUnknown] != teamPlayer};
+		} forEach ((destroyedSites - citiesX) select {sidesX getVariable [_x,sideUnknown] != teamPlayer});
```
Comment (jaj22) [inline A3A/addons/core/functions/init/fn_resourcecheck.sqf:195]: Funny one, rebel resources and factories could be randomly rebuilt.

File: A3A/addons/core/functions/init/fn_initUtilityItems.sqf [hunk 2]
```diff
@@ -33,7 +33,7 @@ private _items = [
 if (LootToCrateRadius == 0) then { _items deleteAt 0 };
 
 if(A3A_hasACE) then {
-    _items pushBack [_medCrate#0, _medCrate#1, "medicalbox", "heal", ["noclear", "move"]],
+    _items pushBack [_medCrate#0, _medCrate#1, "medicalbox", "heal", ["noclear", "move"]];
```
Comment (jaj22) [inline A3A/addons/core/functions/init/fn_initUtilityItems.sqf:36]: Oddly enough, this seems to be valid SQF syntax.

File: A3A/addons/core/functions/REINF/fn_buildMinefield.sqf [hunk 3]
```diff
@@ -110,7 +110,7 @@ if ((_truckX distance _positionTel < 50) and ({alive _x} count units _groupX > 0
 
 	[_groupX, "Patrol_Area", 25, 50, 100, true, _positionTel, true] call A3A_fnc_patrolLoop;
 
-	sleep 30*_quantity;
+	sleep (30 * _quantity);
```
Comment (jaj22) [inline A3A/addons/core/functions/REINF/fn_buildMinefield.sqf:113]: Most likely one to be noticed (although hardly anyone uses the minefield creation, as it's pretty broken further up), as it's a massive increase in minefield build time. Probably fine though? Average minefield is something like 20 mines, so 10 minutes.

**Review by jaj22 — APPROVED** [review]
Should be fine, don't see anything harmful in here.

File: A3A/addons/core/functions/init/fn_initUtilityItems.sqf (hunk 2 above)
Comment (LordGolias) [reply A3A/addons/core/functions/init/fn_initUtilityItems.sqf:36]: indeed, just unusual, so I did it as a pass by.

**Review by Bob-Murphy — APPROVED** [review]

**Pull request merged**